python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -164,6 +164,16 @@ def compute_difference(self, other: "AssetSlice") -> "AssetSlice":
             self._asset_graph_view, self._compatible_subset - other.convert_to_valid_asset_subset()
         )
 
+    def compute_union(self, other: "AssetSlice") -> "AssetSlice":
+        return _slice_from_subset(
+            self._asset_graph_view, self._compatible_subset | other.convert_to_valid_asset_subset()
+        )
+
+    def compute_intersection(self, other: "AssetSlice") -> "AssetSlice":
+        return _slice_from_subset(
+            self._asset_graph_view, self._compatible_subset & other.convert_to_valid_asset_subset()
+        )
+
```

Comment:
hmmm i don't think we need to use the `compute` language here right? This is guaranteed to be fairly fast I think

Reply:
I think at the moment there's one case where things get a bit rough, which is if you have an in-memory time-window subset which only contains the time windows, not the partition keys themselves.

in order to compute the intersection, the current logic has to compute the partition keys for those time windows.

we could theoretically make this more efficient if you have time window to time window intersections (just intersect the time windows themselves), but I think there will always be the case where you have one subset that's encoded in terms of partition keys and another subset encoded with time windows, and it's non-negligible to convert one to the other

Reply:
Ah and then there's the especially-nasty case of multi-partitions definitions, which would require even more cleverness to make efficient in all cases.

Reply:
Got it. Very happy we are doing compute then instead of an operator that might just silently do that!